src/pages/Home/index.jsx

```diff
@@ -1,33 +1,82 @@
-import React, { useContext } from 'react';
+import React, { useContext, useEffect } from 'react';
 import Box from '@material-ui/core/Box';
 import Grid from '@material-ui/core/Grid';
-import { Logo, SmallLogo } from './style';
+import { Logo, SmallLogo, LogoDark, SmallLogoDark } from './style';
 import VideoCard from '../../components/VideoCard';
+import NoResults from '../../components/NotFound/NoResults';
 import logo from '../../utils/assets/logo.gif';
 import smallLogo from '../../utils/assets/small-logo.gif';
+import logoDark from '../../utils/assets/logo-dark.png';
+import smallLogoDark from '../../utils/assets/small-logo-dark.png';
 import { store } from '../../state/store';
-// import videos from '../../utils/youtube-videos-mock.json';
+import youtubeSearch from '../../state/youtube-requests';
+// import videos from '../../utils/mocks/youtube-videos-mock.json';
 
 const HomePage = () => {
   const globalState = useContext(store);
-  const videos =
-    globalState.state.videoList.items === undefined
-      ? []
-      : globalState.state.videoList.items;
+  const { dispatch } = globalState;
+
+  const videos = globalState.state.videoList.items || [];
+
+  useEffect(() => {
+    if (globalState.state.darkMode) {
+      document.body.style.backgroundColor = 'var(--darkgrey)';
+    } else {
+      document.body.style.backgroundColor = 'white';
+    }
+    const fetchVideos = async () => {
+      if (globalState.state.videoList.length === 0) {
+        const queryParams = {
+          type: 'video',
+          q: ' ',
+          maxResults: 25,
+        };
+        const response = await youtubeSearch(queryParams);
+        dispatch({ type: 'search', payload: response.data });
+      }
+    };
+    fetchVideos();
```

Comment:
You're missing to test this section of your component, try to use mock for the youtubeSearch and the dispatch of actions